File: src/main/java/ph/salmon/test/models/Post.java
package ph.salmon.test.models;

public record Post(Integer id, Integer userId, String title, String body) {

    public static class Builder {
        private int id;
        private int userId;
        private String title;
        private String body;

        public Builder id(int id) {
            this.id = id;
            return this;
        }

        public Builder userId(int userId) {
            this.userId = userId;
            return this;
        }

        public Builder title(String title) {
            this.title = title;
            return this;
        }

        public Builder body(String body) {
            this.body = body;
            return this;
        }

        // Метод для создания экземпляра Post
        public Post build() {
            return new Post(id, userId, title, body);
        }
    }
}

File: src/main/kotlin/ph/salmon/test/Post.kt
package ph.salmon.test

data class Post(
    val id: Int,
    val userId: Int,
    val title: String,
    val body: String
)

class PostBuilder {
    private var id: Int = 0
    private var userId: Int = 0
    private var title: String = ""
    private var body: String = ""

    fun setId(id: Int) = apply { this.id = id }
    fun setUserId(userId: Int) = apply { this.userId = userId }
    fun setTitle(title: String) = apply { this.title = title }
    fun setBody(body: String) = apply { this.body = body }

    fun build(): Post {
        return Post(id, userId, title, body)
    }
}